__tests__/shared/utils.ts

```diff
@@ -0,0 +1,61 @@
+import { IAgentOptions, IDIDManager, IResolver, MinimalImportableKey, TAgent } from '../../packages/core/src'
+import { getChainIdForDidEthr, resolveDidOrThrow, mapIdentifierKeysToDoc } from '../../packages/utils/src'
+
+type ConfiguredAgent = TAgent<IResolver & IDIDManager>
+
+export default (testContext: {
+  getAgent: (options?: IAgentOptions) => ConfiguredAgent
+  setup: (options?: IAgentOptions) => Promise<boolean>
+  tearDown: () => Promise<boolean>
+}) => {
+  describe('utilities', () => {
+    let agent: ConfiguredAgent
+
+    beforeAll(async () => {
+      await testContext.setup()
+      agent = testContext.getAgent()
+      return true
+    })
+    afterAll(testContext.tearDown)
+
+    it('should get chainId for ethr did', async () => {
+      const didUrl = 'did:ethr:rinkeby:0xb09b66026ba5909a7cfe99b76875431d2b8d5190'
+      const didDoc = await resolveDidOrThrow(didUrl, {agent})
+      if (didDoc.verificationMethod) {
+        const chainId = getChainIdForDidEthr(didDoc.verificationMethod[0])
+        expect(chainId).toEqual(4)
+      }
+    })
+
+    it('should map identifier keys to did doc', async () => {
+      const account = `0xb09b66026ba5909a7cfe99b76875431d2b8d5190`
+      const did = `did:ethr:0x4:${account}`
+      const controllerKeyId = `metamask-${account}`
+      await agent.didManagerImport({
```

Comment:
Maybe not to be fixed in this PR, but definitely before the next release, so I'm writing this as a reference.

This import seems crucial to the way the web3-kms will behave.
The key metadata, the `controllerKeyId` and `kid` of the key must all align properly for this to work.

Also, if web3 providers change, there should also be some update to the managed DIDs and keys from `@veramo/did-manager` / `@veramo/key-manager`